kube-client/src/client/mod.rs

```diff
@@ -434,26 +432,32 @@ impl Client {
 ///
 /// In either case, present an ApiError upstream.
 /// The latter is probably a bug if encountered.
-fn handle_api_errors(text: &str, s: StatusCode) -> Result<()> {
-    if s.is_client_error() || s.is_server_error() {
+async fn handle_api_errors(res: Response<Body>) -> Result<Response<Body>> {
+    let status = res.status();
+    if status.is_client_error() || status.is_server_error() {
+        // trace!("Status = {:?} for {}", status, res.url());
+        let body_bytes = hyper::body::to_bytes(res.into_body())
+            .await
+            .map_err(Error::HyperError)?;
```

Comment:
the only awkward part about this refactor is that the role of doing the async `to_bytes` call is now awkwardly split between two fns:

- if it's an error; we stream the body in `handle_api_errors`
- if it's not an error, we never streamed it in `handle_api_errors`, and do it later in to two `request_` methods

but i can't really see a way to avoid that, so probably better to leave a comment about it